### PR TITLE
update bug report url to work with latest WC versions

### LINKED ISSUE #67: Undefined wc_version & notice in PHP7.4
On latest WC with PHP7.4 we get 2 notices from this plugin:

```
Undefined index: wc_version
File: /wp-content/plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-admin-menus.php on line 199
Error URL: /wp-admin/post-new.php?post_type=page
Referrer: /wp-admin/post.php?post=1494&action=edit
Trace:
#0 /wp-content/plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-admin-menus.php (222): WC_Beta_Tester_Admin_Menus->construct_ssr()
#1 /wp-content/plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-admin-menus.php (306): WC_Beta_Tester_Admin_Menus->get_github_ticket_url()
#2 /wp-includes/class-wp-hook.php (287): WC_Beta_Tester_Admin_Menus->admin_bar_menus( $wp_admin_bar = WP_Admin_Bar Object )
#3 /wp-includes/class-wp-hook.php (311): WP_Hook->apply_filters( $value = NULL, $args =
Array(
  [0] => WP_Admin_Bar Object
)
)
#4 /wp-includes/plugin.php (544): WP_Hook->do_action( $args = Array( [0] => WP_Admin_Bar Object ) )
#5 /wp-includes/admin-bar.php (97): do_action_ref_array( $tag = admin_bar_menu, $args = Array( [0] => WP_Admin_Bar Object ) )
#6 /wp-includes/class-wp-hook.php (287): wp_admin_bar_render()
#7 /wp-includes/class-wp-hook.php (311): WP_Hook->apply_filters( $value = EMPTY STRING, $args = Array( [0] =>  ) )
#8 /wp-includes/plugin.php (478): WP_Hook->do_action( $args = Array( [0] =>  ) )
#9 /wp-admin/admin-header.php (245): do_action( $tag = in_admin_header )
#10 /wp-admin/edit-form-blocks.php (420): require_once( /wp-admin/admin-header.php )
#11 /wp-admin/post-new.php (72): require( /wp-admin/edit-form-blocks.php )

Trying to access array offset on value of type null
File: /wp-content/plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-admin-menus.php on line 199
Error URL: /wp-admin/post-new.php?post_type=page
Referrer: /wp-admin/post.php?post=1494&action=edit
Trace:
#0 /wp-content/plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-admin-menus.php (222): WC_Beta_Tester_Admin_Menus->construct_ssr()
#1 /wp-content/plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-admin-menus.php (306): WC_Beta_Tester_Admin_Menus->get_github_ticket_url()
#2 /wp-includes/class-wp-hook.php (287): WC_Beta_Tester_Admin_Menus->admin_bar_menus( $wp_admin_bar = WP_Admin_Bar Object )
#3 /wp-includes/class-wp-hook.php (311): WP_Hook->apply_filters( $value = NULL, $args = Array( [0] => WP_Admin_Bar Object ) )
#4 /wp-includes/plugin.php (544): WP_Hook->do_action( $args = Array( [0] => WP_Admin_Bar Object ) )
#5 /wp-includes/admin-bar.php (97): do_action_ref_array( $tag = admin_bar_menu, $args = Array( [0] => WP_Admin_Bar Object ) )
#6 /wp-includes/class-wp-hook.php (287): wp_admin_bar_render()
#7 /wp-includes/class-wp-hook.php (311): WP_Hook->apply_filters( $value = EMPTY STRING, $args = Array( [0] =>  ) )
#8 /wp-includes/plugin.php (478): WP_Hook->do_action( $args = Array( [0] =>  ) )
#9 /wp-admin/admin-header.php (245): do_action( $tag = in_admin_header )
#10 /wp-admin/edit-form-blocks.php (420): require_once( /wp-admin/admin-header.php )
#11 /wp-admin/post-new.php (72): require( /wp-admin/edit-form-blocks.php )
```

### PR DESCRIPTION
**Pull request by rrennick**
Closes #67 

This PR
- Updates minimum WC version to 3.6
- Updates tested to versions to latest WC and WP
- Updates WC API usage to recent WC versions

### Testing

- Navigate to a WC page after switching to this branch
- Click Admin Bar -> WC Beta Tester -> Submit bug report
- This should open a new tab with the issue template pre-populated including the SSR

### CODE REVIEW

File: includes/class-wc-beta-tester-admin-menus.php [hunk 1]
```diff
@@ -152,16 +152,20 @@ private function get_plugins_ssr_info( $active_plugins, $untested_plugins ) {
...
 
 		if ( false === $ssr ) {
 			// When running WC 3.6 or greater it is necessary to manually load the REST API classes.
-			if ( version_compare( WC()->version, '3.6', '>=' ) && ! did_action( 'rest_api_init' ) ) {
+			if ( ! did_action( 'rest_api_init' ) ) {
 				WC()->api->rest_api_includes();
 			}
 
-			if ( ! class_exists( 'WC_REST_System_Status_Controller', false ) ) {
+			if ( ! class_exists( 'WC_REST_System_Status_Controller' ) ) {
```
Comment (claudiosanches) [inline includes/class-wc-beta-tester-admin-menus.php:168]: I think we should keep the autoload off.

File: includes/class-wc-beta-tester-admin-menus.php [hunk 2]
```diff
@@ -184,10 +188,7 @@ protected function construct_ssr() {
 			$ssr = '';
 			foreach ( $response['environment'] as $key => $value ) {
 				$index = $key;
-				// @todo remove this hack after fix schema in WooCommerce, and Claudio never let you folks forget that need to write schema first, and review after every change, schema is the most important part of the REST API.
```
Comment (claudiosanches) [inline includes/class-wc-beta-tester-admin-menus.php:187]: Good old times ahahah

**Comment by kkmuffme** [comment]
Works for me

I just wanted to ask regarding updating the minimum version to 3.6: 
since this plugin allows to go back until version 3.0.0 /wp-admin/plugins.php?page=wc-beta-tester-version-picker there will be issues when we set the minimum version to 3.6?

**Comment by juliaamosova** [comment]
Tested this PR and it works great, thanks @rrennick! 

Before applying the patch, the System Status Report is not being included when opening a ticket using the plugin's UI:

before

After applying the patch, the System Status Report is being included:

after

File: includes/class-wc-beta-tester-admin-menus.php (hunk 1 above)
Comment (rrennick) [reply includes/class-wc-beta-tester-admin-menus.php:168]: @claudiosanches It didn't work for me in master with it off. Can you try it with the `false`?

File: includes/class-wc-beta-tester-admin-menus.php (hunk 2 above)
Comment (rrennick) [reply includes/class-wc-beta-tester-admin-menus.php:187]: That comment is almost worth leaving :).

**Comment by rrennick** [comment]
> there will be issues when we set the minimum version to 3.6?

@kkmuffme Thanks for catching that. We should update the dropdown as well. The current change would only warn about installing this plugin when the installed WC version is older than 3.6.

**Comment by rrennick** [comment]
This needs version bumps in the plugin header.

**Comment by kkmuffme** [comment]
Actually the "WC tested up to" header doesn't make any sense in this plugin, since it's for beta testing new version, which means by definition the WC version will be higher than the "WC tested up to" header value.
I would suggest to remove it

**Review by claudiosanches — APPROVED** [review]

**Comment by rrennick** [comment]
> which means by definition the WC version will be higher than the "WC tested up to" header value.

Thanks for noting this. I'll do some experimenting with that for the next release.

**Pull request merged**